### PR TITLE
[WIP] Add Control Type Information to UIA Tree

### PR DESCRIPTION
**Pull request by chiaramooney**
### Type of Change
- New feature (non-breaking change which adds functionality)

### Why
Add Control Type information to UIA tree to improve accessibility information. 

### What
- Added update for control type information upon fast refresh. 
- Added ability to specify for Text and Edit control types.
- Added DefaultControlType property to all core components to give components a default role. Previously all controls defaulted to Group control type. Now controls default to the following types:
| Component | Control Type |
| -- | -- | 
| View/View-based Controls (Pressable, Touchable*) | Group |
| Button | Button |
| Text | Text |
| Image | Image | 
| Switch | Button | 
| ScrollView | ScrollBar |
| TextInput | Edit | 
| ActivityIndicator | ProgressView | 

## Testing
Tested all controls and confirmed correct types appear in AccessibilityInsights tool. Types can be altered by the developer using the `accessibilityRole` or `role` props and will update upon fast refresh. Control type will return to default value if `accessibilityRole` or `role` is not specified. 
 ###### Microsoft Reviewers: codeflow:open?pullrequest=https://github.com/microsoft/react-native-windows/pull/11876&drop=dogfoodAlpha

### CODE REVIEW

File: vnext/Microsoft.ReactNative/Fabric/Composition/CompositionDynamicAutomationProvider.cpp [hunk 1]
```diff
@@ -135,7 +135,7 @@ HRESULT __stdcall CompositionDynamicAutomationProvider::GetPatternProvider(PATTE
 long GetControlType(const std::string &role) noexcept {
   if (role == "adjustable") {
     return UIA_SliderControlTypeId;
-  } else if (role == "group" || role == "search" || role == "radiogroup" || role == "timer" || role.empty()) {
```
Comment (FalseLobster) [inline vnext/Microsoft.ReactNative/Fabric/Composition/CompositionDynamicAutomationProvider.cpp:138]: The [aria core aam](https://www.w3.org/TR/core-aam-1.1/#mapping_role_table) says that search should map to group and searchbox should map to edit, but search is also unlisted in the official react-native role prop value enum: https://reactnative.dev/docs/accessibility#role

I recommend leaving as is or removing entirely.

File: vnext/Microsoft.ReactNative/Fabric/Composition/CompositionViewComponentView.cpp [hunk 2]
```diff
@@ -1091,6 +1091,9 @@ void CompositionBaseComponentView::updateAccessibilityProps(
       UIA_IsEnabledPropertyId,
       !oldViewProps.accessibilityState.disabled,
       !newViewProps.accessibilityState.disabled);
+
+  winrt::Microsoft::ReactNative::implementation::UpdateUiaProperty(
```
Comment (FalseLobster) [inline vnext/Microsoft.ReactNative/Fabric/Composition/CompositionViewComponentView.cpp:1095]: I genuinely don't know if this is the right call. Generally speaking I don't think ATs expect control type to change and aren't likely to listen to prop change notifications for ControlType.  I feel like I saw MSDN guidance on this, but all I can find in this moment is [a bit more ambiguous](https://learn.microsoft.com/en-us/windows/win32/winauto/uiauto-propertiesoverview). I don't think there's much harm in this and may be the right call given how react-native can make this property come in late due to a rerender or something, but figured I should at least speculate openly on it :)

**Review by FalseLobster — APPROVED** [review]

File: vnext/Microsoft.ReactNative/Fabric/Composition/CompositionViewComponentView.cpp (hunk 2 above)
Comment (chiaramooney) [reply vnext/Microsoft.ReactNative/Fabric/Composition/CompositionViewComponentView.cpp:1095]: Makes sense! I'll do a bit of reading and check what behavior we set on Paper!

Comment (chiaramooney) [reply vnext/Microsoft.ReactNative/Fabric/Composition/CompositionViewComponentView.cpp:1095]: Looks like on Paper we did check for and make updates to accessibilityRole in the UpdateProperty() functions. Maybe for now we leave as is to match the Paper behavior? But definitely a good callout. It's possible we were doing the wrong thing before and should keep that on our radar.

Comment (FalseLobster) [reply vnext/Microsoft.ReactNative/Fabric/Composition/CompositionViewComponentView.cpp:1095]: Sounds good to me

**Review by FalseLobster — APPROVED** [review]

**Pull request merged**